src/loader.js

```diff
@@ -0,0 +1,91 @@
+let assume = require('assume');
+let debug = require('debug')('taskcluster:loader');
+let Promise = require('promise');
+
+function loader(componentDirectory, requiredVirtualComponents) {
+  /**
+   * If present, this is a list of component names that a loader
+   * must define.  These are likely things like configuration values
+   */
+  if (requiredVirtualComponents) {
+    assume(requiredVirtualComponents).is.array();
+  } else {
+    requiredVirtualComponents = [];
+  }
```

Comment:
Also can I have: (just steal it from here)
```js
var TopoSort = require('topo-sort');
var tsort = new TopoSort();
for (let c of requiredVirtualComponents) {
  tsort.add(c, []);
}
_.forEach(componentDirectory, (def, name) => {
  tsort.add(name, def.requires);
});
try {
  components = tsort.sort();
} catch (err) {
  throw new Error("Expected component dependencies to form a DAG");
}

if (!componentDirectory.graphviz) {
  componentDirectory.graphviz = {requires: [], setup: () => {
    console.log('digraph G {');
    for(let c of components) {
      console.log('"' + c + '";');
    }
    _.forEach(componentDirectory, ({requires}, name) => {
      for(let c of requires) {
        console.log('"' + name + '" -> "' + c + '"[dir=back];');
      }
    });
    console.log('}');
  }};
}
```
Then we get a defaukt graphviz target that prints dependencies between components in the loader...